### PR TITLE
Introduce tests & CI

### PR DESCRIPTION
**Pull request by dy**
This PR introduces a showcase for baseline tests (pilot).
Now CI is available for gl-components (yay).
That should reduce complexity of test coverage for gl-based traces in plotly.

Also that makes gl-component tests look way less messy, in standard tape- fashion.
@etpinard @archmoj please see [test.js](https://github.com/gl-vis/regl-scatter2d/blob/palette/test.js), hope that will make further dev simpler and fix regressions (seems that palette work may cause some).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,8 +18,7 @@
   ],
   "browserify": {
     "transform": [
-      "glslify",
-      "bubleify"
```
Comment (etpinard) [inline package.json:21]: Hmm. You no longer convert es6 -> es5?

**Comment by etpinard** [comment]
`npm test` is working for me locally :tada:

File: package.json (hunk 1 above)
Comment (dy) [reply package.json:21]: bubleify does not do async/await https://github.com/Rich-Harris/buble/pull/125, and seems there is no more maintenance.
I wished there was a way for browserify to indicate this transform for main export, but ignore that for running tests.
Probably I should include babelify. Or just simply rewrite the thing in ES5, that is not a lot of hassle. Rollup maybe?

**Comment by dy** [comment]
@etpinard there is a way to bundle the package as well, without sacrificing the dependencies. That is not as elegant as just browserify with `--ignore`, but the latter is unfortunately broken for that use-case.

That adds:
* ES5 compilation on prepublish
* tree shaking, if required
* Removes glslify and other browserify runtime transforms (plotly bundling gets faster)
* Keeps shared dependencies instead of packing them to bundle (`ignore` option)
* Enables target browser versions via `browserslist` option https://github.com/browserslist/browserslist

File: package.json [hunk 2]
```diff
@@ -2,10 +2,12 @@
...
   "version": "3.0.6",
   "description": "Scatter2d plot built with regl",
-  "main": "./index.js",
+  "main": "./bundle.js",
   "scripts": {
-    "test": "budo test",
-    "build": "browserify test.js -g bubleify | indexhtmlify | metadataify | github-cornerify | mobilify > gh-pages/no-snapping.html"
+    "test": "node test",
+    "test-browser": "budo test -- -t glslify",
+    "prepublishOnly": "npm run buikd",
```
Comment (etpinard) [inline package.json]: `build`

**Comment by etpinard** [comment]
Well, that's annoying. How many more bytes is that going to result in plotly.js? Oh well, maybe plotly.js is already too big to care about bundling glslify twice.

**Comment by dy** [comment]
@etpinard glslify does not get bundled, that just inlines glsl dependencies to a bundle. Now in plotly.js glslify does inlining whenever we run plotly build, but with that PR glslify is not required anymore, because glsl is inlined already, so plotly building gets faster. 

Moreover, right now plotly building also runs bubleify, that performs the same transpilation internally. With that PR we cut that as well, by providing transpiled code.

More to that, with this PR glslify is not a production, but dev dependency, meaning that plotly.js package as a dependency is faster to run with browserify, and the component is quicker to install.

**Comment by dy** [comment]
@etpinard should we proceed with that?

**Comment by etpinard** [comment]
Looks like I was completely wrong in https://github.com/gl-vis/regl-scatter2d/pull/15#issuecomment-439156668, building plotly.js off the latest commit on this branch results in a slightly **smaller** bundle:

image


That said, looks like this PR here breaks a bunch of plotly.js tests:

- `(plotly.js) $ npm run test-image-gl2d` barfs, there are still some non ES5 tokens in the regl-scatter2d `bundle.js` that don't get along with our old nw.js image test server.
- `(plotly.js) $ npm run test-jasmine -- --tags=gl` spits out a bunch of

image

errors.

- `(plotly.js) $ npm run test-syntax` gives:

image

**Comment by dy** [comment]
Oh Lord, I've found that. regl-splom uses wrong regl-scatter2d entry, pointing to not a bundle.
@etpinard can you please check tests with https://github.com/gl-vis/regl-splom/pull/10
They run fine on my side.

**Comment by etpinard** [comment]
plotly.js tests are green, :dancer: 

Thanks @dy !

**Pull request merged**